mbf_costmap_nav/src/mbf_costmap_nav/free_pose_search.cpp

```diff
@@ -59,6 +59,17 @@ FreePoseSearch::FreePoseSearch(costmap_2d::Costmap2DROS& costmap, const SearchCo
   compare_strategy_ = std::move(compare_strategy.value_or(euclidean_compare));
 }
 
+bool FreePoseSearch::isPoseValid(const unsigned char cost) const
+{
+  return cost != costmap_2d::LETHAL_OBSTACLE && cost != costmap_2d::INSCRIBED_INFLATED_OBSTACLE &&
+         cost != costmap_2d::NO_INFORMATION;
```

Comment:
Q: Why do we consider states with cost inscribed as valid but poses with cost inscribed as invalid? Shouldn't they have the same meaning?

Reply:
When you check the pose, you query costmap the cost as if the center of the robot is on that cell. That means you are checking the pose of the robot. If the cost is `INSCRIBED`, the robot is for sure in collision and we don't need to verify the footprint (it will be LETHAL for sure).
To check the state, you query all cells that belongs to the footprint (border and inside) and then get the max cost. Therefore, if the border is `INSCRIBED`, it is fine. The problem arises only if a cell is LETHAL.